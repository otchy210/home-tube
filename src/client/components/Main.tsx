import { CircularProgress } from '@material-ui/core';
import { useEffect } from 'react';
import { useState } from 'react';
import { fetchJson } from './common/fetchJson';
import I18nContext, { I18nContextProps, Translations } from './common/I18nContext';
import SettingsContext, { SettingsContextProps } from './common/SettingsContext';
import Base from './layout/Base';

const Main: React.FC<{}> = () => {
    const [loading, setLoading] = useState(true);
    const [settings, setSettings] = useState<SettingsContextProps>({});
    const [i18n, setI18n] = useState<I18nContextProps>({});
    useEffect(() => {
        const locale = 'ja';
        (async () => {
            const settings = await fetchJson<SettingsContextProps>('/settings.json');
            setSettings(settings);
            const translations = await fetchJson<Translations>(`/i18n/${locale}.json`);
            setI18n({ locale, translations });
            setLoading(false);
        })();
    }, []);
    return (
        <>
            {loading ? (
                <CircularProgress />
            ) : (
                <SettingsContext.Provider value={settings}>
                    <I18nContext.Provider value={i18n}>
                        <Base />
                    </I18nContext.Provider>
                </SettingsContext.Provider>
            )}
        </>
    );
};

export default Main;
