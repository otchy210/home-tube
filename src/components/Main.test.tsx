import React from 'react';
import { render } from '@testing-library/react';
import Main from './Main';

describe('Main', () => {
    it('renders properly', () => {
        const { container } = render(<Main></Main>);
        expect(container).toMatchSnapshot();
    });
});
