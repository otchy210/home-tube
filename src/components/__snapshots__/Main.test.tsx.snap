// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders properly 1`] = `
<div>
  <div>
    main
  </div>
</div>
`;
